### PR TITLE
add sql maintenance window rule

### PR DESCRIPTION
**Pull request by BrunoReboul**
The SQL Maintenance Window Constraint will scan GCP SQL instances and validate that a maintenance window has been set.
[
If you do not specify a maintenance window, disruptive updates can happen at any time](https://cloud.google.com/sql/docs/mysql/instance-settings#maintenance-window-2ndgen)

### CODE REVIEW

File: validator/test/fixtures/sql_maintenance_window/assets/data.json [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                  "ipv4Enabled":false,
+                  "privateNetwork":"projects/test-project/global/networks/default"
+               },
+               "kind":"sql#settings",
+               "locationPreference":{  
+                  "kind":"sql#locationPreference",
+                  "zone":"us-central1-a"
+               },
+               "maintenanceWindow":{  
+                  "day":0,
```
Comment (AdrienWalkowiak) [inline validator/test/fixtures/sql_maintenance_window/assets/data.json]: Is this real data? From what I can tell, when you don't set a maintenance window for you SQL instance, the maintenanceWindow key is abstent from the data.

File: policies/templates/gcp_sql_maintenance_window_v1.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        kind: GCPSQLMaintenanceWindowConstraintV1
+        plural: gcpsqlmaintenancewindowconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties: {}
+  targets:
+    validation.gcp.forsetisecurity.org:
+      rego: | #INLINE("validator/sql_maintenance_window.rego")
+        #
+        # Copyright 2018 Google LLC
```
Comment (AdrienWalkowiak) [inline policies/templates/gcp_sql_maintenance_window_v1.yaml]: nit: update to 2019

File: validator/sql_maintenance_window.rego [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#
+# Copyright 2018 Google LLC
```
Comment (AdrienWalkowiak) [inline validator/sql_maintenance_window.rego]: nit: update to 2019

File: validator/sql_maintenance_window_test.rego [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#
+# Copyright 2018 Google LLC
```
Comment (AdrienWalkowiak) [inline validator/sql_maintenance_window_test.rego]: nit: update to 2019

File: validator/test/fixtures/sql_maintenance_window/assets/data.json (hunk 1 above)
Comment (BrunoReboul) [reply validator/test/fixtures/sql_maintenance_window/assets/data.json]: Now with real data. SQL first generation do not support maintenance window and are filtered out. for 2nd generation the maintenanceWindow key always exit. day is set to 0 when no maintenance window defined.

File: validator/sql_maintenance_window.rego (hunk 3 above)
Comment (BrunoReboul) [reply validator/sql_maintenance_window.rego]: done

File: validator/sql_maintenance_window_test.rego (hunk 4 above)
Comment (BrunoReboul) [reply validator/sql_maintenance_window_test.rego]: Done

File: policies/templates/gcp_sql_maintenance_window_v1.yaml (hunk 2 above)
Comment (BrunoReboul) [reply policies/templates/gcp_sql_maintenance_window_v1.yaml]: done

**Comment by BrunoReboul** [comment]
@googlebot I fixed it.

**Pull request closed without merging**